File: scripts/bankspace.py
#!/usr/bin/env python3

# determine the amount of free space at the end of each bank in an oracles rom.

import sys


with open(sys.argv[1], 'rb') as f:
    rom = f.read()

for i in range(0x40):
    run = 0
    for j in range(0x4000):
        if rom[i*0x4000+j] in (0, i):
            run += 1
        else:
            run = 0
    print(hex(i), hex(run))

File: scripts/bankspace.lua
#!/usr/bin/env lua

-- determine the amount of free space at the end of each bank in an oracles
-- rom.

NUM_BANKS = 0x40
BANK_SIZE = 0x4000

local f = io.open(arg[1], 'rb')
local rom = f:read('*a')
f:close()

for bank = 0, NUM_BANKS - 1 do
    local run = 0

    for i = 1, BANK_SIZE do
        local byte = string.byte(rom, bank * BANK_SIZE + i)

        -- ages banks are padded with zeroes.
        -- seasons banks are padded with the bank number.
        if byte == 0 or byte == bank then
            run = run + 1
        else
            run = 0
        end
    end

    print(string.format('%02x %x', bank, run))
end
